### PR TITLE
Ports Custom Vendors from /tg/station

### LINKED ISSUE #2056: [Bounty] Port custom vendors from TG.
One commissary is not enough for one station. This should help cargo in selling items they order for a ton of money to the crew and SOMEHOW be SOMEWHAT profitable. 

Main PR you need to port: https://github.com/tgstation/tgstation/pull/46240
Fix that allows to actually load stuff: https://github.com/tgstation/tgstation/pull/46351
I guess it is a fix for selling stacks of items: https://github.com/tgstation/tgstation/pull/46538

PR these, make them merged and I will accept your claim.

### PR DESCRIPTION
**Pull request by Valtosin**
## About The Pull Request

Ports custom vendors from /tg/station and also accidentally fixes some missing spans in _vending.dm.

## Why It's Good For The Game

Now you can sell this precious crowbar only just for 999 credits. Buy NOW!

## Changelog
:cl:
add: custom vendor + price tagger port from /tg/station
add: adds price tagger, refill and circuit for custom vendor in vendomat
add: autolathe can too produce price tagger and custom refill
fix: some missing spans in _vending.dm
/:cl:

This fixes #2056

### CODE REVIEW

**Comment by imw-Oculus** [comment]
Fix your merge error, please. This is good shit.

**Comment by Valtosin** [comment]
oh shit it still unmerged lol

fixed

**Comment by St0rmC4st3r** [comment]
image
Looks good and functional. Everything looks like a glass, but it is not our problem. I guess it is ready for vote/merge.

**Comment by ike709** [comment]
> Looks good and functional. Everything looks like a glass, but it is not our problem. 

Why isn't it our problem?

**Comment by St0rmC4st3r** [comment]
> > Looks good and functional. Everything looks like a glass, but it is not our problem.
> 
> Why isn't it our problem?

/tg/ has it too. From original PR up to this day. It would be nice to get it fixed though.

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -834,6 +895,209 @@ GLOBAL_LIST_EMPTY(vending_products)
...
+			var/base64
+			var/price = 0
+			for(var/obj/T in contents)
+				if(T.name == O)
+					price = T.custom_price
+					if(!base64)
+						if(base64_cache[T.type])
+							base64 = base64_cache[T.type]
+						else
+							base64 = icon2base64(icon(T.icon, T.icon_state))
```
Comment (St0rmC4st3r) [inline code/modules/vending/_vending.dm:961]: I guess the problem lies somewhere here.

**Comment by St0rmC4st3r** [comment]
@Valtosin please fix the icon thingy

**Comment by Valtosin** [comment]
Problem is here:
https://github.com/BeeStation/BeeStation-Hornet/blob/88286fd551ec6b44515028849fbb4f34d497dfb7/tgui/packages/tgui/interfaces/Vending.js#L26-L28
Should work now:
image

**Comment by eLeCtrOssSnake** [comment]
I want this now, merge!!

**Review by eLeCtrOssSnake — APPROVED** [review]
Looks like an old custom vendor with some changes to work with reworked vendors. Seems fishy.

**Comment by ike709** [comment]
If anything breaks blame @St0rmC4st3r

**Pull request merged**